### ISSUE TITLE
Minified react error #31 when trying to edit dashboard annotations

### ISSUE THREAD

**Issue by pprkut** [grafana/grafana#47872]
**What happened**:

We have an existing dashboard with annotations configured. If I go to `Dashboard settings` and then `Annotations` I see an error:

```
Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Btype%2C%20uid%7D for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```

**What you expected to happen**:

Be able to edit/add annotations

**How to reproduce it (as minimally and precisely as possible)**:

[dashboard.json.txt](https://github.com/grafana/grafana/files/8512007/dashboard.json.txt)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.6
- Data source type & version: InfluxDB 5.0.0
- OS Grafana is installed on: CentOS 7
- User OS & Browser: Slackware Linux - Firefox 98.0.2
- Grafana plugins: -
- Others: -

**Comment by joshhunt** [comment]
I'm [not able to reproduce ](https://play.grafana.org/d/atdfvVw7z/47875-old-table-panel?orgId=1&from=now-15m&to=now&editview=annotations)this using a different data source, or using the JSON you supplied, but the link in the exception hints to there being a problem with how we display the name of the data source in the annotations list.

**Comment by joshhunt** [comment]
I notice that in another dashboard it displays the data source UI for the data source, which isn't great 
![image](https://user-images.githubusercontent.com/46142/164245798-ab96e234-7669-4b58-af2f-da4bbab58139.png)

**Comment by pprkut** [comment]
@joshhunt It's a provisioned datasource and a provisioned dashboard. It's a bit older, but we regularly update the dashboard based on grafana's dashboard JSON. We have it provisioned to multiple grafana instances and the problem happens on all of them.

If it helps, we have one other dashboard where we use annotations and there everything works fine, so it seems to be something specific to this one.

**Comment by JoaoSilvaGrafana** [comment]
Hey! Thanks for raising this issue, the error is fixed with 8.5.0 which is going to be released today :) The problem that Josh mentioned with the IDs is still there though and I have created a separate issue for it and will investigate soon.

**Comment by pprkut** [comment]
Can confirm this no longer happens on 8.5.0. Thanks! :)
